tests/unittests/InterpreterTest.cpp

```diff
@@ -896,3 +896,84 @@ TEST(Interpreter, nonLinearClassifier) {
     EXPECT_NEAR(RH.at({0, std::get<2>(tests[i])}), 1.0, 0.2);
   }
 }
+
+/// Generate images in two classes. A "line" is labeled as 0
+/// and a "cross" is labeled as 1
+void generateImageData(Tensor &images, Tensor &labels, unsigned numSamples) {
+  auto L = labels.getHandle<size_t>();
+  auto Im = images.getHandle<>();
+  images.zero();
+
+  for (size_t i = 0; i < numSamples; i++) {
+    L.at({i, 0}) = i % 2;
```

Comment:
One way to make this code more readable is to write:

bool isLine = i % 2; 


L.at({i, 0}) = isLine; 


 if (isLine) {
...